browser/net/brave_oauth_protocol_handler.h

```diff
@@ -0,0 +1,28 @@
+/* Copyright (c) 2020 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#ifndef BRAVE_BROWSER_NET_BRAVE_OAUTH_PROTOCOL_HANDLER_H_
+#define BRAVE_BROWSER_NET_BRAVE_OAUTH_PROTOCOL_HANDLER_H_
+
+#include <string>
+
+#include "base/optional.h"
+#include "content/public/browser/web_contents.h"
+#include "ui/base/page_transition_types.h"
+#include "url/gurl.h"
+
+namespace oauth {
+
+void HandleOauthProtocol(const GURL& url,
```

Comment:
Would you mind adding an explanation of what this function is supposed to do?

Reply:
It is worth mentioning that it actually always makes an async call (i.e. `PostTask` to UI) no matter of which thread or sequence calls it